### PR TITLE
Move global unprotected temporary exceptions to selected features on build

### PR DESCRIPTION
**Pull request by laghee**
**Asana Task/Github Issue:** https://app.asana.com/0/1200890834746050/1202346295795542/f

## Description
We shouldn't be globally excluding certain non-explicitly-privacy-protecting or otherwise-unrelated-to-web-pages features when we mitigate for breakage. This change distributes the domain exceptions individually to only the features we should disable in these cases.

#### Reference
- [Config Reviewer Documentation](https://app.asana.com/0/1200890834746050/1204443212791216/f)
- [Config Maintainer Documentation](https://app.asana.com/0/1200890834746050/1200573250322769/f)
- [Feature Implementer Documentation](https://app.asana.com/0/1200890834746050/1201498956177210/f)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -100,11 +101,20 @@ function addExceptionsToUnprotected (exceptions) {
 }
 
 const listData = JSON.parse(fs.readFileSync(`${LISTS_DIR}/${unprotectedListName}`))
-defaultConfig.unprotectedTemporary = listData.exceptions
-
-addExceptionsToUnprotected(defaultConfig.unprotectedTemporary)
+addExceptionsToUnprotected(listData.exceptions)
 addExceptionsToUnprotected(defaultConfig.features.contentBlocking.exceptions)
 
+const notExcluded = ['adClickAttribution', 'appTrackerProtection', 'autofill', 'duckPlayer', 'incontextSignup',
```
Comment (jonathanKingston) [inline index.js:107]: `notExcluded` could be read either way as not excluded from this treatment or not an exclusion.
Could we consider using a name that adds a little more clarity like `excludedFeaturesFromUnprotectedTemporaryExceptions` (maybe there's something less wordy it's late here ha)

File: index.js [hunk 2]
```diff
@@ -100,11 +101,20 @@ function addExceptionsToUnprotected (exceptions) {
...
+    'incrementalRolloutTest', 'networkProtection', 'newTabContinueSetUp', 'voiceSearch', 'windowsPermissionUsage',
+    'windowsWaitlist', 'windowsDownloadLink']
+for (const exception of listData.exceptions) {
+    for (const key of Object.keys(defaultConfig.features)) {
+        if (!(notExcluded.includes(key))) {
+            defaultConfig.features[key].exceptions = defaultConfig.features[key].exceptions.concat(exception)
```
Comment (jonathanKingston) [inline index.js:113]: `concat` can accept an array, we can remove the `const exception of listData.exceptions` and make this line:
```suggestion
            defaultConfig.features[key].exceptions = defaultConfig.features[key].exceptions.concat(listData.exceptions)
```

File: index.js (hunk 1 above)
Comment (laghee) [reply index.js]: Didn't have any kind of brainstorm on shortening that aside from `Temp` for `Temporary` assuming we want to keep all the context 🤷‍♀️

File: index.js [hunk 3]
```diff
@@ -100,11 +101,18 @@ function addExceptionsToUnprotected (exceptions) {
 }
 
 const listData = JSON.parse(fs.readFileSync(`${LISTS_DIR}/${unprotectedListName}`))
-defaultConfig.unprotectedTemporary = listData.exceptions
-
-addExceptionsToUnprotected(defaultConfig.unprotectedTemporary)
+addExceptionsToUnprotected(listData.exceptions)
 addExceptionsToUnprotected(defaultConfig.features.contentBlocking.exceptions)
 
+const excludedFeaturesFromUnprotectedTempExceptions = ['adClickAttribution', 'appTrackerProtection', 'autofill', 'duckPlayer', 'incontextSignup',
```
Comment (SlayterDev) [inline index.js:107]: This is a nit but could we break this list out to one entry per line for readability?

Comment (SlayterDev) [inline index.js:107]: Let's maybe also drop a comment above this line to explain what the list is.

**Review by SlayterDev — COMMENTED** [review]
Thank you @laghee! I've verified the generated output looks correct. I've added a couple nits here just to improve the readability.

**Review by SlayterDev — APPROVED** [review]
LGTM!

**Pull request merged**